src/features/transfer/useTokenTransfer.ts

```diff
@@ -117,336 +86,106 @@ async function executeTransfer({
+    if (!activeAccountAddress) throw new Error('No active account found for origin chain');
+
+    const warpCore = getWarpCore();
+
+    const isCollateralSufficient = await warpCore.isDestinationCollateralSufficient(
+      tokenAmount,
+      destination,
+    );
+    if (!isCollateralSufficient) {
+      toast.error('Insufficient collateral on destination for transfer');
+      throw new Error('Insufficient destination collateral');
+    }
 
     addTransfer({
-      activeAccountAddress,
       timestamp: new Date().getTime(),
-      status,
-      route: tokenRoute,
-      params: values,
+      status: TransferStatus.Preparing,
+      origin,
+      destination,
+      originTokenAddressOrDenom: originToken.addressOrDenom,
+      destTokenAddressOrDenom: connection.token.addressOrDenom,
+      sender: activeAccountAddress,
+      recipient,
+      amount,
     });
 
-    const executeParams: ExecuteTransferParams<any, any> = {
-      weiAmountOrId,
-      originProtocol,
-      destinationDomainId,
-      recipientAddress,
-      tokenRoute,
+    updateTransferStatus(transferIndex, (transferStatus = TransferStatus.CreatingTxs));
+
+    const txs = await warpCore.getTransferRemoteTxs(
+      tokenAmount,
+      destination,
       activeAccountAddress,
-      activeChain,
-      updateStatus: (s: TransferStatus) => {
-        status = s;
-        updateTransferStatus(transferIndex, s);
-      },
-      sendTransaction: transactionFns[originProtocol].sendTransaction,
-    };
+      recipient,
+    );
 
-    let transferTxHash: string;
-    let msgId: string | undefined;
-    if (isWarpRoute(tokenRoute)) {
-      ({ transferTxHash, msgId } = await executeHypTransfer(executeParams));
-    } else if (isIbcRoute(tokenRoute)) {
-      ({ transferTxHash } = await executeIbcTransfer(executeParams));
-    } else {
-      throw new Error('Unsupported route type');
+    const hashes: string[] = [];
+    for (const tx of txs) {
+      updateTransferStatus(transferIndex, (transferStatus = txCategoryToStatuses[tx.category][0]));
+      const { hash, confirm } = await sendTransaction({
+        tx: tx.transaction,
+        chainName: origin,
+        activeChainName: activeChain.chainName,
+        providerType: tx.type,
+      });
+      updateTransferStatus(transferIndex, (transferStatus = txCategoryToStatuses[tx.category][1]));
+      const receipt = await confirm();
+      const description = toTitleCase(tx.category);
+      logger.debug(`${description} transaction confirmed, hash:`, receipt.transactionHash);
+      toastTxSuccess(`${description} transaction sent!`, receipt.transactionHash, origin);
+      hashes.push(hash);
     }
 
-    updateTransferStatus(transferIndex, (status = TransferStatus.ConfirmedTransfer), {
-      originTxHash: transferTxHash,
-      msgId,
-    });
+    // TODO
+    // const msgId = tryGetMsgIdFromTransferReceipt(transferReceipt);
```

Comment:
how do you think you'll go about doing this? rpc with the last tx hash or keep track of the last tx receipt?

Reply:
My hope is that the tx receipt for cosmos will, like evm and solana, contain the tx logs and then double hope that those logs contain the message id. Otherwise I'm not sure how we'll do this for cosmos